Add immutable feature to text field of `WholeList`

pott-cho opened the pull request:
Close: #289

To support cases where fields should not be modified after creation, the `disabled` option was added instead of using `readonly`.
The `disabled` option was chosen to allow the application of different CSS styles compared to components pre defined with `readonly`.

With this change, the immutable property is now properly applied to the `WholeList` > `Text` field.
For fields such as IDs that should not be modified once created, the field's title can be added to `immutable_fields`, making the corresponding Text field appear as disabled only in the edit modal.
If other UI components such as Select or Checkbox require immutable support in the future, this commit can be used as a reference for further extension.

One concern is that the Edit and Create modals currently cannot be controlled separately from the outside.
The same modal is used in both cases, and for this update, the small difference was handled by adding `immutable_fields` to the `WholeList`.
However, if there is a future need for significantly different UI or behavior between the Edit and Create modals, it may be necessary to decouple the WholeList and its input modal and define separate modals for editing and creation.

sehkone commented:
In `WholeList`, this change makes it look like users are expected to specify which fields are immutable after creation. However, I suggest applying this behavior only to the necessary items in `InputConfig`. This way, I believe we can minimize the chance of user errors, such as typos in field names, and limit the feature to a minimal set of items, rather than applying it universally.

If there are any issues that makes my suggestion impractical, please let me know; this isn't a thoroughly verified idea yet.

pott-cho commented:
> In `WholeList`, this change makes it look like users are expected to specify which fields are immutable after creation. However, I suggest applying this behavior only to the necessary items in `InputConfig`. This way, I believe we can minimize the chance of user errors, such as typos in field names, and limit the feature to a minimal set of items, rather than applying it universally.
> 
> If there are any issues that makes my suggestion impractical, please let me know; this isn't a thoroughly verified idea yet.

Adding the is_immutable property to InputConfig to make certain fields immutable requires specifying the option every time TextConfig is used, since there is currently no way to provide a default value in this structure. To avoid this repetition, I initially implemented the behavior at the WholeList level, where defaults can be set more easily.

As you pointed out, handling immutability directly within InputConfig would certainly keep the frontary code cleaner, but I’m a bit concerned about the potential impact this change might have on other modules. I'd like to hear your thoughts on this.

sehkone commented:
> Adding the is_immutable property to InputConfig to make certain fields immutable requires specifying the option every time TextConfig is used, since there is currently no way to provide a default value in this structure. To avoid this repetition, I initially implemented the behavior at the WholeList level, where defaults can be set more easily.

I understand your concern. In fact, we could set default values by adding `new` methods like the following:

```rust
impl TextConfig {
    fn new(...) {
        ...
        self.is_immutable = false;
        ... 
    }

    fn new_immutable(...) {
        ...
        self.is_immutable = true;
    }
}
```
I am not saying that we should necessarily implement these methods—this is just an example. Even without them, I don't think it's a big issue to set the `is_immutable` value for every single item. It may be less convenient, but it's less error-prone than your approach—at least from a user's (i.e., developer's) perspective.

pott-cho commented:
@sehkone
I added an Immutable field to TextConfig to allow manual configuration. Since using the new method would compromise consistency with other similar configs, I implemented the manual setup approach as you suggested. Thank you.

pott-cho commented:
> Since users have to be informed of this change, could you update the changelog?

I have updated the changelog.

sehkone commented:
It looks good to me. I haven’t tested it yet though, so I’ll leave the rest of the review to others.

pott-cho commented:
I rebased the commits to help with the next review.

sophie-cluml commented:
@pott-cho Could you share me which part of the code does the `user_input` refer to, in your commit message please?

pott-cho commented:
@sophie-cluml 
By user_input, I was referring to the user_input module (user_input.rs). All the changes in this commit were made as a result of adding the immutable option to user_input.rs.

sophie-cluml commented:
Could you rebase please?

pott-cho commented:
@sophie-cluml 
I've completed the rebase.
